compiler/noirc_evaluator/src/ssa/plonky2_gen/asm_writer.rs

```diff
@@ -172,6 +174,30 @@ impl Display for U32TargetDisplay {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+pub struct DebugTraceSourcePoint {
+    pub file: String,
+    pub line_number: usize,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DebugTraceAsmListIndexRange {
+    pub start: usize,
+    pub end: Option<usize>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct DebugTraceList {
+    pub list: Vec<String>,
+    pub source_map: HashMap<DebugTraceSourcePoint, Vec<DebugTraceAsmListIndexRange>>,
+}
+
+impl DebugTraceList {
+    pub fn new() -> DebugTraceList {
+        DebugTraceList { list: vec![], source_map: HashMap::new() }
+    }
+}
```

Comment:
These structures could be in a separate file, removing the need to prefix them all with DebugTrace.

Reply:
Moved and renamed: https://github.com/blocksense-network/noir/pull/96/commits/a08ab5cb9b5820e44a086e185369e3ac9d6094f9 https://github.com/blocksense-network/noir/pull/96/commits/a9da947584f1d37f8e43941f1db0d3874b9f39ed https://github.com/blocksense-network/noir/pull/96/commits/1776fdc4abc47bf8fad074194ac4fb33a11e2af0 https://github.com/blocksense-network/noir/pull/96/commits/f15cdb1e5bdcc48dc95b6365d2d9ada81e63d46d